rhel/vex/parser.go

```diff
@@ -0,0 +1,450 @@
+	}
+}
+
+func (pc *productCache) Get(c *csaf.CSAF, productID string) *csaf.Product {
+	if p, ok := pc.cache[productID]; ok {
+		return p
+	}
+	p := c.ProductTree.FindProductByID(productID)
+	pc.cache[productID] = p
+	return p
+
+}
+
+func NewCreator(vulnName, vulnLink string, c *csaf.CSAF, pc *productCache, rc *repoCache) *creator {
+	return &creator{
+		vulnName:       vulnName,
+		vulnLink:       vulnLink,
+		uniqueVulnsIdx: make(map[string]int),
+		c:              c,
+		pc:             pc,
+		rc:             rc,
+	}
+}
+
+type creator struct {
+	vulnName, vulnLink string
+	uniqueVulnsIdx     map[string]int
+	fixedVulns         []claircore.Vulnerability
+	c                  *csaf.CSAF
+	pc                 *productCache
+	rc                 *repoCache
+}
+
+func (c *creator) knownAffectedVulnerabilities(ctx context.Context, v csaf.Vulnerability, protoVulnFunc func() *claircore.Vulnerability) ([]*claircore.Vulnerability, error) {
+	// Known affected
+	out := []*claircore.Vulnerability{}
+	for _, pc := range v.ProductStatus["known_affected"] {
+		rel := c.c.ProductTree.Relationships.FindRelationship(pc, "default_component_of")
+		if rel == nil {
+			// Should never get here, error in data
+			zlog.Debug(ctx).
+				Str("product:package", pc).
+				Msg("could not find a relationship for product:package")
+			continue
+		}
+		if strings.HasPrefix(rel.ProductRef, "kernel") {
+			// We don't want to ingest kernel advisories as
+			// containers have no say in the kernel.
+			continue
+		}
+		repoProd := c.pc.Get(c.c, rel.RelatesToProductRef)
+		if repoProd == nil {
+			// Should never get here, error in data
+			zlog.Warn(ctx).
+				Str("prod", rel.RelatesToProductRef).
+				Msg("could not find product in product tree")
+			continue
+		}
+		cpeHelper, ok := repoProd.IdentificationHelper["cpe"]
+		if !ok {
+			zlog.Warn(ctx).
+				Str("prod", rel.RelatesToProductRef).
+				Msg("could not find cpe helper type in product")
+			continue
+		}
+
+		// pkgName will be overriden if we find a valid pURL
+		pkgName := rel.ProductRef
+		compProd := c.pc.Get(c.c, rel.ProductRef)
+		if compProd == nil {
+			// Should never get here, error in data
+			zlog.Warn(ctx).
+				Str("pkg", rel.ProductRef).
+				Msg("could not find package in product tree")
+			continue
+		}
+		// It is possible that we will not find a pURL, in that case
+		// the package.Name will be reported as-is.
+		purlHelper, ok := compProd.IdentificationHelper["purl"]
+		if ok {
+			purl, err := packageurl.FromString(purlHelper)
+			switch {
+			case err != nil:
+				zlog.Warn(ctx).
+					Str("purl", purlHelper).
+					Err(err).
+					Msg("could not parse PURL")
+			default:
+				pkgName = purl.Name
+			}
+			if purl.Type != packageurl.TypeRPM || purl.Namespace != "redhat" {
+				// Just ingest advisories that are Red Hat RPMs, this will
+				// probably change down the line when we consolidate updaters.
+				continue
+			}
+		}
+		if strings.HasPrefix(pkgName, "kernel") {
```

Comment:
why check again?

Reply:
This is checking the `pkgName` which could be distinct if it's an `oci` purl (although that probably negates it from being a kernel advisory. I'll remove this check